functests/2_performance_update/kubelet.go

```diff
@@ -0,0 +1,177 @@
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/discovery"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/mcps"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+)
+
+var _ = Describe("[ref_id: 45487][performance]additional kubelet arguments", func() {
+	var profile *performancev2.PerformanceProfile
+	var workerRTNodes []corev1.Node
+	var performanceMCP string
+
+	testutils.BeforeAll(func() {
+		workerRTNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred())
+
+		profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		performanceMCP, err = mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Verify that worker and performance MCP have updated state equals to true
+		for _, mcpName := range []string{testutils.RoleWorker, performanceMCP} {
+			mcps.WaitForCondition(mcpName, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		}
+
+	})
+	BeforeEach(func() {
+		if discovery.Enabled() && testutils.ProfileNotFound {
+			Skip("Discovery mode enabled, performance profile not found")
+		}
+	})
+	Context("Additional kubelet arguments", func() {
+		It("[test_id:45488]Test performance profile annotation for changing multiple kubelet settings", func() {
+			sysctlAnnotations := map[string]string{
+				"kubeletconfig.experimental": "{\"allowedUnsafeSysctls\":[\"net.core.somaxconn\",\"kernel.msg*\"],\"systemReserved\":{\"memory\":\"300Mi\"},\"kubeReserved\":{\"memory\":\"768Mi\"},\"imageMinimumGCAge\":\"3m\"}",
+			}
+			profile.SetAnnotations(sysctlAnnotations)
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			profiles.UpdateWithRetry(profile)
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+			By("Waiting when mcp finishes updates")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+			kubeletArguments := []string{"/bin/bash", "-c", "ps -ef | grep kubelet | grep config"}
+			for _, node := range workerRTNodes {
+				kubeletConfig, err := nodes.GetKubeletConfig(&node)
+				Expect(err).ToNot(HaveOccurred())
+				sysctlsValue := kubeletConfig.AllowedUnsafeSysctls
+				Expect(kubeletSliceValues(sysctlsValue, "net.core.somaxconn")).To(BeTrue())
```

Comment:
this
```
https://onsi.github.io/gomega/#containelementelement-interface
```
is probably more expressive -> better error message by default on failure (here and in the similar instances all around)

Reply:
In the latest patch i have addressed this.